lib/kvdb/ikvdb.c

```diff
@@ -3050,11 +3048,9 @@ ikvdb_kvs_cursor_read_copy(
         return 0;
 
     kvs_cursor_key_copy(cur, keybuf, keybuf_sz, NULL, key_len);
-    if (valbuf) {
-        err = kvs_cursor_val_copy(cur, valbuf, valbuf_sz, NULL, val_len);
-        if (ev(err))
-            return err;
-    }
+    err = kvs_cursor_val_copy(cur, valbuf, valbuf_sz, NULL, val_len);
```

Comment:
Would this try to decompress the value even if valbuf is NULL?

Reply:
Looks like if valbuf==nil it will try to decompress into cur->kci_buf

Reply:
Right. I didn't really want to change this up too much because with 3.0, I would like to get rid of plain read, seek, and seek range which would hopefully remove the need for the buffer on the cursor.

Reply:
I could probably skip this if buf and val_out are null though. Would have to check with testing.

Reply:
Can confirm. I will make this part of my PR.